### PR TITLE
Story/temp accuracy map/911

### PR DESCRIPTION
**Pull request by andrea-williams**
https://wps-pr-1285.apps.silver.devops.gov.bc.ca/morecast

Note: this doesn't work in Test env bc WFWX API isn't quick enough in responding to request for weather data - it times out.

### CODE REVIEW

**Review by tessmgood — APPROVED** [review]
V rad! Conditional approval: 
Small, weird thing — when I toggle between RH and temp using the select field, an active state persists after the switch which makes the 'view accuracy map for' label hard to read (blue on blue) unless I click away, then the label reverts to white. Your call: we can fix it later, I want to tighten up that whole footer area UI wise. Someone is very likely to click a station or the map which immediately fixes it.

**Comment by andrea-williams** [comment]
> V rad! Conditional approval:
> Small, weird thing — when I toggle between RH and temp using the select field, an active state persists after the switch which makes the 'view accuracy map for' label hard to read (blue on blue) unless I click away, then the label reverts to white. Your call: we can fix it later, I want to tighten up that whole footer area UI wise. Someone is very likely to click a station or the map which immediately fixes it.

Should the label just always be white?

**Comment by tessmgood** [comment]
> > V rad! Conditional approval:
> > Small, weird thing — when I toggle between RH and temp using the select field, an active state persists after the switch which makes the 'view accuracy map for' label hard to read (blue on blue) unless I click away, then the label reverts to white. Your call: we can fix it later, I want to tighten up that whole footer area UI wise. Someone is very likely to click a station or the map which immediately fixes it.
> 
> Should the label just always be white?

Yes :)

File: api/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # NOTE:
 # This Dockerfile is for local development only!
 
-FROM python:3.8-buster
+FROM python:3.8.12-buster
```
Comment (conbrad) [inline api/Dockerfile:4]: Why are we pinning a minor version?

File: api/app/wildfire_one/wfwx_api.py [hunk 2]
```diff
@@ -128,7 +128,7 @@ async def get_detailed_stations(time_of_interest: datetime):
                     temperature=daily.get('temperature'),
                     relative_humidity=daily.get('relativeHumidity'))
                 record_type = daily.get('recordType').get('id')
-                if record_type == 'ACTUAL':
+                if record_type in ['ACTUAL', 'MANUAL']:
```
Comment (conbrad) [inline api/app/wildfire_one/wfwx_api.py:131]: What is manual?

File: web/package.json [hunk 3]
```diff
@@ -60,7 +60,7 @@
     "start": "CI=true react-scripts start",
     "start:cypress": "export $(cat .env.cypress | xargs) && react-scripts -r @cypress/instrument-cra start",
     "build": "react-scripts build",
-    "test": "react-scripts test",
+    "test": "react-scripts test --transformIgnorePatterns \"node_modules/(?!ol)/\"",
```
Comment (conbrad) [inline web/package.json:63]: Why is this necessary?

File: web/src/features/fireWeather/components/AccuracyColorLegend.tsx [hunk 4]
```diff
@@ -10,31 +20,30 @@ const useStyles = makeStyles({
...
-    marginLeft: '40px'
+    textAlign: 'center'
   },
   leftLabel: {
     height: '15px',
     width: '80px',
     fontSize: '10px',
     color: 'white',
-    marginLeft: '4px'
+    justifyContent: 'flex-start'
```
Comment (conbrad) [inline web/src/features/fireWeather/components/AccuracyColorLegend.tsx:32]: Nice 😎

File: web/src/features/fireWeather/components/AccuracyColorLegend.tsx [hunk 5]
```diff
@@ -44,52 +53,107 @@ const useStyles = makeStyles({
...
+    backgroundColor: darkOrangeColor,
+    borderTop: '2px solid #FFFFFF',
+    borderRight: '2px solid #FFFFFF',
+    borderBottom: '2px solid #FFFFFF',
+    borderRadius: '0px 2px 2px 0px',
+    height: 27,
+    width: 32
+  },
+  darkestRed: {
+    backgroundColor: '#720505',
```
Comment (conbrad) [inline web/src/features/fireWeather/components/AccuracyColorLegend.tsx:119]: I like how the rest of the classes referenced named colors, can we do the same for the remaining ones?

File: web/src/features/fireWeather/components/AccuracyVariablePicker.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    color: 'white'
+  },
+  '& .MuiFormLabel-root.Mui-focused': {
+    color: 'white'
+  }
+})
+
+interface Props {
+  selectedWxVariable: AccuracyWeatherVariableEnum
+  changeHandler: any
```
Comment (conbrad) [inline web/src/features/fireWeather/components/AccuracyVariablePicker.tsx:26]: Lint is complaining about the any, I think this can be a function that accepts an event and returns void.

File: web/src/features/fireWeather/components/maps/WeatherMap.tsx [hunk 7]
```diff
@@ -44,24 +40,51 @@ const source = new olSource.XYZ({
 
 const zoom = 6
 
+const rhPointStyleFunction = (feature: any) => {
```
Comment (conbrad) [inline web/src/features/fireWeather/components/maps/WeatherMap.tsx:43]: This any we don't know the type for, I think an eslint exception comment to suppress the warning would help.

File: web/src/features/fireWeather/components/maps/WeatherMap.tsx [hunk 8]
```diff
@@ -44,24 +40,51 @@ const source = new olSource.XYZ({
...
+  return new Style({
+    image: new CircleStyle({
+      radius: computeRHAccuracySize(feature.values_),
+      fill: new Fill({ color: rhPointColor }),
+      stroke: computeStroke(rhPointColor)
+    })
+  })
+}
+
+const tempPointStyleFunction = (feature: any) => {
```
Comment (conbrad) [inline web/src/features/fireWeather/components/maps/WeatherMap.tsx:54]: Similar to above.

File: web/src/features/fireWeather/components/maps/WeatherMap.tsx [hunk 9]
```diff
@@ -93,7 +116,7 @@ const WeatherMap = ({
     },
     [dispatch]
   )
-
+  console.log(stations)
```
Comment (conbrad) [inline web/src/features/fireWeather/components/maps/WeatherMap.tsx]: console.log snuck in

File: web/src/features/fireWeather/components/maps/stationAccuracy.test.ts [hunk 10]
```diff
@@ -1,155 +1,509 @@
...
+
+    it('should return the largest scale index when observed RH is much lower than forecasted RH', () => {
+      expect(computeRHScaleIndex(100, 10)).toEqual(rhColorScale.length - 1)
+    })
+
+    it('should return a scale index of 0 when observed RH is much greater than forecasted RH', () => {
+      expect(computeRHScaleIndex(10, 100)).toEqual(0)
+    })
+
+    it('should return a scale index of neutralIndex+1 when observed RH is 4 percentage points lower than forecasted RH', () => {
```
Comment (conbrad) [inline web/src/features/fireWeather/components/maps/stationAccuracy.test.ts:66]: Minor: I thought it was 3 percentage points? Maybe this has changed or I'm remembering wrong.

**Review by conbrad — CHANGES_REQUESTED** [review]

File: api/app/wildfire_one/wfwx_api.py (hunk 2 above)
Comment (andrea-williams) [reply api/app/wildfire_one/wfwx_api.py:131]: Manual is where someone goes into WFWX API and manually overrides whatever values were automatically reported by the sensors at a weather station (I think this only ever happens when there's something wrong with a sensor). There were a lot of them recently in WFWX API and it was breaking things

File: api/Dockerfile (hunk 1 above)
Comment (andrea-williams) [reply api/Dockerfile:4]: Because poetry.lock specifies that the Python version must be `>=3.8.6`, but `3.8-buster` has a lower minor version, so the local Docker doesn't work unless we specify the only available minor version >= .6

File: web/package.json (hunk 3 above)
Comment (andrea-williams) [reply web/package.json:63]: In WeatherMap we import styling components from OpenLayers, and Jest breaks unless we override `ol` node_modules because of problems with how/when the package is compiled

File: api/Dockerfile (hunk 1 above)
Comment (conbrad) [reply api/Dockerfile:4]: Ah this is for local docker. We'd lose higher minor updates by pinning this but for local dev it's low risk.

File: web/src/features/fireWeather/components/maps/stationAccuracy.test.ts (hunk 10 above)
Comment (andrea-williams) [reply web/src/features/fireWeather/components/maps/stationAccuracy.test.ts:66]: Each step is 3 percentage points. So that means if observed RH is 3 percentage points lower than forecasted RH, the forecast is considered "correct" and the neutral colour will be shown. This test is checking for what happens when the forecast is considered "slightly wrong", as in the map marker should not be a neutral colour.

File: web/src/features/fireWeather/components/maps/WeatherMap.tsx (hunk 9 above)
Comment (andrea-williams) [reply web/src/features/fireWeather/components/maps/WeatherMap.tsx]: Deleted :)

**Review by JulianForeman — APPROVED** [review]

**Review by conbrad — APPROVED** [review]

**Review by Sybrand — APPROVED** [review]
